### PR TITLE
Added changes to enable Ceph toolbox

### PR DESCRIPTION
**Pull request by kesavanvt**
The ocsinitialization is updated to enable the ceph toolbox in deployer namespace

Signed-off-by: kesavan <kvellalo@redhat.com>

### CODE REVIEW

File: controllers/managedocs_controller.go [hunk 1]
```diff
@@ -592,6 +615,18 @@ func (r *ManagedOCSReconciler) reconcileDMSPrometheusRule() error {
...
 
+func (r *ManagedOCSReconciler) reconcileOCSInitialization() error {
+	r.Log.Info("Reconciling OCSInitialization")
+
+	if err := r.get(r.ocsInitialization); err == nil {
+		r.ocsInitialization.Spec.EnableCephTools = true
+		if err = r.Client.Update(r.ctx, r.ocsInitialization); err != nil {
+			return err
+		}
+	}
```
Comment (nb-ohad) [inline controllers/managedocs_controller.go:626]: A couple of points:
1. I think we do want to fail the reconcile if the error is no `errors.IsNotFound` because it means that the deployer has a genuine issue retrieving the object that we would like to identify and fix (for example RBAC issues)
2. I will consider returning the underlaying error (of a generic Get or Create operation) without encapsulating it with a context a bad practice. let's add a `fmt.ErrorF` with a more meaningful error" *(you can still attach the original error message to the new one)
3. If we don't find the resource (IsNotFound) I think it will be helpful to document that in the log as a Warning or an Info

File: controllers/managedocs_controller_test.go [hunk 2]
```diff
@@ -805,6 +811,20 @@ var _ = Describe("ManagedOCS controller", func() {
...
+
+				ocsInitKey := utils.GetResourceKey(ocsInit)
+
+				// Wait for the spec changes to be reverted
+				Eventually(func() bool {
+					Expect(k8sClient.Get(ctx, ocsInitKey, ocsInit)).Should(Succeed())
+					return ocsInit.Spec.EnableCephTools
+				}, timeout, interval).Should(Equal(true))
+			})
+		})
```
Comment (nb-ohad) [inline controllers/managedocs_controller_test.go:827]: Missing a test case to validate that if the field is changed we revert the change (should be very similar to the create test and can even assume that ocsInitialization was already created and the field was set by the prev test

**Review by nb-ohad — CHANGES_REQUESTED** [review]

File: controllers/managedocs_controller.go [hunk 3]
```diff
@@ -163,6 +166,14 @@ func (r *ManagedOCSReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
 		),
 	)
+	ocsInitializationPredicates := builder.WithPredicates(
+		predicate.NewPredicateFuncs(
+			func(meta metav1.Object, _ runtime.Object) bool {
+				name := meta.GetName()
+				return name == ocsInitializationName
+			},
+		),
+	)
```
[the remark this answers is not in the document]
Comment (kesavanvt) [reply controllers/managedocs_controller.go:176]: ocs operator creates the OCSInitialization instance called `ocsinit` so the above predicate reacts only to `ocsinit` and why do we react to any OCSInitialization resource?

[the remark this answers is not in the document]
Comment (nb-ohad) [reply controllers/managedocs_controller.go:176]: You can argue the same for the storagecluster resource. The fact of the matter is that we do not try to differentiate on OCSInitialization that we should reconcile and ones that we shouldn't. Not like other resources (e.g. secrets) where we have no interest in touching instances that does not relates to the deployer

[the remark this answers is not in the document]
Comment (kesavanvt) [reply controllers/managedocs_controller.go:176]: Deployer owns the storagecluster resource. In case of OCSInitialization , the instance(`ocsinit`) is created by OCS operator. The deployment of toolbox depends only by the spec from `ocsinit` instance. So even if there is a case when another instance of OCSInitialization is created why it is actually need to trigger the reconciliation on creation of it?.

[the remark this answers is not in the document]
Comment (kesavanvt) [reply controllers/managedocs_controller.go]: I have made the changes as discussed.

File: controllers/managedocs_controller_test.go [hunk 4]
```diff
@@ -805,6 +811,38 @@ var _ = Describe("ManagedOCS controller", func() {
...
+		When("the ocsInitialization resource is modified", func() {
+			It("should revert the changes in ocsInitialization to enable ceph toolbox", func() {
+				ocsInit := ocsInitializationTemplate.DeepCopy()
+				ocsInitKey := utils.GetResourceKey(ocsInit)
+				Consistently(func() error {
+					return k8sClient.Get(ctx, ocsInitKey, ocsInit)
+				}, timeout, interval).Should(Succeed())
```
Comment (nb-ohad) [inline controllers/managedocs_controller_test.go:834]: Why Consistently? did you mean Eventually? if you just need Eventually on a resource you can use `utils.WaitForResource`

**Review by nb-ohad — CHANGES_REQUESTED** [review]

File: controllers/managedocs_controller_test.go (hunk 4 above)
Comment (kesavanvt) [reply controllers/managedocs_controller_test.go]: My bad, simple `Expect` satisfied the need.

**Review by nb-ohad — APPROVED** [review]

**Comment by nb-ohad** [comment]
@JohnStrunk Please merge
